### PR TITLE
ci: update to Go 1.15.4 and alpine:3.12

### PR DESCRIPTION
**Pull request by mikemorris**
Is there any substantial reason we can't/shouldn't bump to Go 1.15.x for the Consul 1.9 release series? Did I miss bumping the version in any files?

### CODE REVIEW

**Comment by mikemorris** [comment]
In the go-test-api `TestAPI_ClientTLSOptions`, several tests are failing with `x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0`

**Comment by mikemorris** [comment]
It looks like maybe static validation of time is more strict in 1.15, not sure if this simply means two of these failing tests could be removed now...
```
TestSession_Apply_BadTTL
    session_endpoint_test.go:929: incorrect error message: Session TTL '10z' invalid: time: unknown unit "z" in duration "10z"
```
```
TestConfigUtil_Values
    config_test.go:85: (case 3) err: 1 error(s) decoding:
        
        * error decoding 'duration': time: invalid duration "nope"

**Comment by rboyer** [comment]
@mikemorris I fixed the two tests in question. The error messages are mostly the same, except that now the inputs are wrapped in double quotes now so our assertions were slightly incorrect.

**Comment by rboyer** [comment]
Now there's this fun one with multiple occurrences: 
```
 CONT  TestSetupHTTPServer_HTTP2
    http_test.go:196: err: Get "https://127.0.0.1:38331/echo": x509: certificate relies on legacy Common Name field, use SANs or temporarily enable Common Name matching with GODEBUG=x509ignoreCN=0
```

**Comment by rboyer** [comment]
might be good to wait until `1.15.4` for https://github.com/golang/go/issues/42138

**Comment by rboyer** [comment]
I also regenerated some of the TLS test data files so they don't anger the updated tls stack in Go.

**Comment by mikemorris** [comment]
Yeaaaa, `x509: certificate relies on legacy Common Name field` is my biggest concern with moving to 1.15, it feels like a change we might want to look into a little more closely, particularly implications on service mesh things...

**Comment by mikemorris** [comment]
> might be good to wait until `1.15.4` for [golang/go#42138](https://github.com/golang/go/issues/42138)

Sooo, clock skew is potentially not fun, and [we just cut a 1.8.5 Docker release with `alpine:3.12`](https://github.com/hashicorp/docker-consul/blob/f49c996c29119a14f5e58f09432fcdfe496390c3/0.X/Dockerfile) - I'm slightly unclear on if the issue is only present when running a compiled binary on an affected version of alpine or if the issue is which version of alpine was used to compile the binary.

/cc @alvin-huang

**Comment by rboyer** [comment]
> Yeaaaa, `x509: certificate relies on legacy Common Name field` is my biggest concern with moving to 1.15, it feels like a change we might want to look into a little more closely, particularly implications on service mesh things...

Service mesh doesn't use the Common Name, preferring instead the newer SubjectAlternativeName (SAN) field. The former is singly valued and the latter is multi valued.

**Comment by rboyer** [comment]
This lets you easily display the common name and subject alternative name fields from the CLI:
```
$ openssl x509  -noout -in server.crt -subject -ext subjectAltName
```

File: tlsutil/config.go [hunk 1]
```diff
@@ -920,11 +920,6 @@ func (c *Configurator) wrapALPNTLSClient(dc, nodeName, alpnProto string, conn ne
 		return nil, err
 	}
 
-	if cs := tlsConn.ConnectionState(); !cs.NegotiatedProtocolIsMutual {
```
Comment (mkeeler) [inline tlsutil/config.go:923]: Just curious but why did this need removal?

Comment (mikemorris) [reply tlsutil/config.go:923]: Deprecated in 1.15, it's always `true` now (and gave a compiler error when checking it) https://golang.org/pkg/crypto/tls/

Comment (mikemorris) [reply tlsutil/config.go:923]: If I understand correctly, it would now fail to establish a connection at all if in the case wheree it would be false.

Comment (mkeeler) [reply tlsutil/config.go:923]: :thumbsup:

**Review by mkeeler — APPROVED** [review]

**Pull request merged**